config/initializers/101-lograge.rb

```diff
@@ -109,26 +109,28 @@ def unsubscribe(component_name, subscriber)
             }
 
             if data = (Thread.current[:_method_profiler] || event.payload[:timings])
-              sql = data[:sql]
-
-              if sql
+              if sql = data[:sql]
                 output[:db] = sql[:duration] * 1000
                 output[:db_calls] = sql[:calls]
               end
 
-              redis = data[:redis]
-
-              if redis
+              if redis = data[:redis]
                 output[:redis] = redis[:duration] * 1000
                 output[:redis_calls] = redis[:calls]
               end
 
-              net = data[:net]
-
-              if net
+              if net = data[:net]
                 output[:net] = net[:duration] * 1000
                 output[:net_calls] = net[:calls]
               end
+
+              # MethodProfiler.stop is called after this lambda, so the delta
+              # must be computed here.
+              if start_heap_live_slots = data[:__start_heap_live_slots]
+                if heap_live_slots = GC.stat[:heap_live_slots] - start_heap_live_slots
```

Comment:
I don't think we need this conditional here because the odds of the difference being 0 is quite unlikely.